Decred Wallet/Features/Transactions/TransactionsViewController.swift

```diff
@@ -0,0 +1,239 @@
+            self.txTableView.reloadData()
+            self.refreshControl.endRefreshing()
+        }
+
+        let selectedFilterIndex = self.txFilterDropDown.selectedItemIndex
+        let currentFilterItem = self.txFilters[safe: selectedFilterIndex] ?? DcrlibwalletTxFilterAll
+
+        let selectedSortIndex = self.txSortOrderDropDown.selectedItemIndex
+        let sortOrderNewerFirst = self.txSortOrders[safe: selectedSortIndex] ?? true
+
+        if let txs = WalletLoader.shared.firstWallet?.transactionHistory(offset: 0, count: 0, filter: currentFilterItem, newestFirst: sortOrderNewerFirst) {
+            self.allTransactions = txs
+        }
+    }
+}
+
+extension TransactionsViewController: DcrlibwalletTxAndBlockNotificationListenerProtocol {
+    func onBlockAttached(_ walletID: Int, blockHeight: Int32) {
+        // not relevant to this VC
+    }
+
+    func onTransaction(_ transaction: String?) {
+        let tx = try! JSONDecoder().decode(Transaction.self, from: (transaction!.utf8Bits))
+
+        if self.allTransactions.contains(where: { $0.hash == tx.hash }) {
+            // duplicate notification, tx is already being displayed in table
+            return
+        }
+
+        if self.newTxHashes.firstIndex(of: tx.hash) != nil {
+            return
+        }
+
+        self.newTxHashes.append(tx.hash)
+        self.allTransactions.insert(tx, at: 0)
+
+        // Save hash for this tx as last viewed tx hash.
+        Settings.setStringValue(tx.hash, for: DcrlibwalletLastTxHashConfigKey)
+
+        DispatchQueue.main.async {
+            self.reloadTxsForCurrentFilter()
+        }
+    }
+
+    func onTransactionConfirmed(_ walletID: Int, hash: String?, blockHeight: Int32) {
+        // all tx statuses will be updated when table rows are reloaded.
+         DispatchQueue.main.async {
+            self.txTableView.reloadData()
+        }
+    }
+}
+
+extension TransactionsViewController: UITableViewDataSource, UITableViewDelegate {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return self.allTransactions.count
+    }
+
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        return TransactionTableViewCell.height()
+    }
+
+    func tableView(_ tableView: UITableView, willDisplay cell: UITableViewCell, forRowAt indexPath: IndexPath) {
+        let tx = self.allTransactions[indexPath.row]
+        if let newTxHashIndex = self.newTxHashes.firstIndex(of: tx.hash) {
+            cell.blink()
+            self.newTxHashes.remove(at: newTxHashIndex)
+        }
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = self.txTableView.dequeueReusableCell(withIdentifier: TransactionTableViewCell.identifier) as! TransactionTableViewCell
+
+        var frame = cell.frame
+        frame.size.width = self.txTableView.frame.size.width
+        cell.frame = frame
```

Comment:
Don't think these are necessary. Table view cells typically assume the parent table view width.